newlib/libc/sys/vita/error.c

```diff
@@ -0,0 +1,205 @@
+		case SCE_NET_ERROR_EROFS:
+			return EROFS;
+		case SCE_NET_ERROR_EMLINK:
+			return EMLINK;
+		case SCE_NET_ERROR_EPIPE:
+			return EPIPE;
+		case SCE_NET_ERROR_EDOM:
+			return EDOM;
+		case SCE_NET_ERROR_ERANGE:
+			return ERANGE;
+		case SCE_NET_ERROR_EAGAIN:
+			return EAGAIN;
+// these values are the same as EAGAIN
+//		case SCE_NET_ERROR_EWOULDBLOCK:
+//			return EWOULDBLOCK;
+		case SCE_NET_ERROR_EINPROGRESS:
+			return EINPROGRESS;
+		case SCE_NET_ERROR_EALREADY:
+			return EALREADY;
+		case SCE_NET_ERROR_ENOTSOCK:
+			return ENOTSOCK;
+		case SCE_NET_ERROR_EDESTADDRREQ:
+			return EDESTADDRREQ;
+		case SCE_NET_ERROR_EMSGSIZE:
+			return EMSGSIZE;
+		case SCE_NET_ERROR_EPROTOTYPE:
+			return EPROTOTYPE;
+		case SCE_NET_ERROR_ENOPROTOOPT:
+			return ENOPROTOOPT;
+		case SCE_NET_ERROR_EPROTONOSUPPORT:
+			return EPROTONOSUPPORT;
+		case SCE_NET_ERROR_ESOCKTNOSUPPORT:
+			return ESOCKTNOSUPPORT;
+		case SCE_NET_ERROR_EOPNOTSUPP:
+			return EOPNOTSUPP;
+		case SCE_NET_ERROR_EPFNOSUPPORT:
+			return EPFNOSUPPORT;
+		case SCE_NET_ERROR_EAFNOSUPPORT:
+			return EAFNOSUPPORT;
+		case SCE_NET_ERROR_EADDRINUSE:
+			return EADDRINUSE;
+		case SCE_NET_ERROR_EADDRNOTAVAIL:
+			return EADDRNOTAVAIL;
+		case SCE_NET_ERROR_ENETDOWN:
+			return ENETDOWN;
+		case SCE_NET_ERROR_ENETUNREACH:
+			return ENETUNREACH;
+		case SCE_NET_ERROR_ENETRESET:
+			return ENETRESET;
+		case SCE_NET_ERROR_ECONNABORTED:
+			return ECONNABORTED;
+		case SCE_NET_ERROR_ECONNRESET:
+			return ECONNRESET;
+		case SCE_NET_ERROR_ENOBUFS:
+			return ENOBUFS;
+		case SCE_NET_ERROR_EISCONN:
+			return EISCONN;
+		case SCE_NET_ERROR_ENOTCONN:
+			return ENOTCONN;
+		case SCE_NET_ERROR_ESHUTDOWN:
+			return ESHUTDOWN;
+		case SCE_NET_ERROR_ETOOMANYREFS:
+			return ETOOMANYREFS;
+		case SCE_NET_ERROR_ETIMEDOUT:
+			return ETIMEDOUT;
+		case SCE_NET_ERROR_ECONNREFUSED:
+			return ECONNREFUSED;
+		case SCE_NET_ERROR_ELOOP:
+			return ELOOP;
+		case SCE_NET_ERROR_ENAMETOOLONG:
+			return ENAMETOOLONG;
+		case SCE_NET_ERROR_EHOSTDOWN:
+			return EHOSTDOWN;
+		case SCE_NET_ERROR_EHOSTUNREACH:
+			return EHOSTUNREACH;
+		case SCE_NET_ERROR_ENOTEMPTY:
+			return ENOTEMPTY;
+		case SCE_NET_ERROR_EUSERS:
+			return EUSERS;
+		case SCE_NET_ERROR_EDQUOT:
+			return EDQUOT;
+		case SCE_NET_ERROR_ESTALE:
+			return ESTALE;
+		case SCE_NET_ERROR_EREMOTE:
+			return EREMOTE;
+		case SCE_NET_ERROR_ENOLCK:
+			return ENOLCK;
+		case SCE_NET_ERROR_ENOSYS:
+			return ENOSYS;
+		case SCE_NET_ERROR_EIDRM:
+			return EIDRM;
+		case SCE_NET_ERROR_EOVERFLOW:
+			return EOVERFLOW;
+		case SCE_NET_ERROR_EILSEQ:
+			return EILSEQ;
+		case SCE_NET_ERROR_ENOTSUP:
+			return ENOTSUP;
+		case SCE_NET_ERROR_ECANCELED:
+			return ECANCELED;
+		case SCE_NET_ERROR_EBADMSG:
+			return EBADMSG;
+		case SCE_NET_ERROR_ENODATA:
+			return ENODATA;
+		case SCE_NET_ERROR_ENOSR:
+			return ENOSR;
+		case SCE_NET_ERROR_ENOSTR:
+			return ENOSTR;
+		case SCE_NET_ERROR_ETIME:
+			return ETIME;
+		default:
+			return EINVAL;
```

Comment:
I would suggest making sure that `sce_errno < 0`, and otherwise return `0` (or the original value?).

Reply:
return original would be good. but imo if we return another value, make confused who only read opengroup document.

maybe we need another static variable `sce_errno` like `errno`

Reply:
I thought this, just will return original value.

Reply:
> also only can come in already `error value < 0` so don't need recheck

What do you mean? `sce_errno` is signed, so it can perfectly be negative if this function is misused or if the value passed to it isn't checked properly.

Reply:
this code replaced
```
if (res < 0) {
  return res & 0xFF;
}
```
to
```
if (res < 0) {
   return __vita_sce_errno_to_errno(res);
}
```
it means, first sce_errno must be already less then 0.
also this function is internal function and that name has `sce_errno_to`.
I think we can handle it & could be avoid wrong using.